label_studio/io_storages/localfiles/models.py

```diff
@@ -0,0 +1,79 @@
+"""This file and its contents are licensed under the Apache License 2.0. Please see the included NOTICE for copyright information and LICENSE for a copy of the license.
+"""
+import logging
+import json
+import re
+
+from pathlib import Path
+from django.db import models, transaction
+from django.utils.translation import gettext_lazy as _
+from django.conf import settings
+
+from io_storages.base_models import ImportStorage, ImportStorageLink, ExportStorage, ExportStorageLink
+from tasks.serializers import AnnotationSerializer
+
+logger = logging.getLogger(__name__)
+url_scheme = 'https'
+
+
+class LocalFilesMixin(models.Model):
+    path = models.TextField(
+        _('path'), null=True, blank=True,
+        help_text='Local path')
+    regex_filter = models.TextField(
+        _('regex_filter'), null=True, blank=True,
+        help_text='Regex for filtering objects')
+    use_blob_urls = models.BooleanField(
+        _('use_blob_urls'), default=False,
+        help_text='Interpret objects as BLOBs and generate URLs')
+
+
+class LocalFilesImportStorage(ImportStorage, LocalFilesMixin):
+
+    def iterkeys(self):
+        path = Path(self.path)
+        regex = re.compile(str(self.regex_filter)) if self.regex_filter else None
+        for file in path.rglob('*'):
+            if file.is_file():
+                key = file.name
+                if regex and not regex.match(key):
+                    logger.debug(key + ' is skipped by regex filter')
+                    continue
+                yield file.name
+
+    def get_data(self, key):
+        if self.use_blob_urls:
+            # include self-hosted links pointed to local resources via /data/filename?d=<path/to/local/dir>
+            path = str(Path(self.path) / key)
+            return {settings.DATA_UNDEFINED_NAME: f'/data?d={path}'}
```

Comment:
As far as I remember we already have /data endpoint, it's better to make something like /data/local-files

Reply:
Changed to `/data/local-files`